GuideForNewDevelopers.md

```diff
@@ -0,0 +1,579 @@
+- [Guide for New Developers](#guide-for-new-developers)
```

Comment:
I would put a proper title instead of starting with bullet points right away. Something like this:
# Guide for New Developers
[Table of Contents](#0table-of-contents)
(and then remove the title on level 1 from below, after table of contents)